### PR TITLE
list all categories of an entry; upper categories in sidebar widget

### PR DESCRIPTION
**Pull request by mgisbers**
range over all categories in blog list and correct display name to be category

upper categories in sidebar widget to lookalike tags.

### CODE REVIEW

File: layouts/_default/list.html [hunk 1]
```diff
@@ -42,7 +42,15 @@ <h2><a href="{{ .Permalink }}">{{ .Title }}</a></h2>
                                           {{ end }}
                                           {{ if isset .Params "categories" }}
                                           {{ if gt (len .Params.categories) 0 }}
-                                          in <a href="{{ "categories/" | relURL }}{{ index .Params.categories 0 | urlize | lower }}">{{ index .Params.categories 0 }}</a>
+                                          in
```
Comment (ryanfox1985) [inline layouts/_default/list.html:45]: I would like to not repeat code, could you replace the two ranges into one?

Like this example: https://stackoverflow.com/questions/21305865/golang-separating-items-with-comma-in-template

File: layouts/partials/widgets/categories.html [hunk 2]
```diff
@@ -12,7 +12,7 @@ <h3 class="panel-title">{{ i18n "categoriesTitle" }}</h3>
             {{ $current := . }}
             {{ range $name, $items := .Site.Taxonomies.categories }}
             <li{{ if eq $current.RelPermalink ($name | urlize | lower | printf "/categories/%s/") }} class="active"{{ end }}>
-                <a href="{{ "categories/" | relURL }}{{ $name | urlize | lower }}">{{ $name }} ({{ len $items }})</a>
+                <a href="{{ "categories/" | relURL }}{{ $name | urlize | lower }}">{{ $name | upper }} ({{ len $items }})</a>
```
Comment (ryanfox1985) [inline layouts/partials/widgets/categories.html:15]: Good catch!

**Review by ryanfox1985 — COMMENTED** [review]
I'm not sure if we have a post with multiples categories already, if there isn't could you add one? Thanks!

File: layouts/_default/list.html (hunk 1 above)
Comment (mgisbers) [reply layouts/_default/list.html:45]: Thanks for pointing at that. changed.

**Review by ryanfox1985 — APPROVED** [review]
LGTM

**Pull request merged**